manifests/install.pp

```diff
@@ -37,21 +36,73 @@
       'AIX'   => concat(['--ignoreos'],$install_options),
       default => $install_options
     }
-    $_package_options = {
+
+    # package provider does not provide 'versionable'
+    package { $::puppet_agent::package_name:
+      ensure          => 'present',
       provider        => 'rpm',
       source          => "/opt/puppetlabs/packages/${package_file_name}",
       install_options => $_install_options,
     }
   } elsif $::operatingsystem == 'Solaris' and $::operatingsystemmajrelease == '10' {
```

Comment:
There are major differences in how we do the upgrade between 10 and 11. I believe 11 uses upgrade, instead of uninstall/reinstall. It's quite possible our upgrade process doesn't restart the service until after upgrade.